cv_bridge/CMakeLists.txt

```diff
@@ -12,6 +12,8 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra)
 endif()
 
+add_compile_options(-DHAVE_SNPRINTF)
```

Comment:
I believe this is to treat the following warning:

```
warning C4996: '_vsnprintf': This function or variable may be unsafe. Consider using _vsnprintf_s instead.
```

Could you please replace the vnsprintf calls with vsnprintf_s instead of adding this compile option?

Reply:
The error that I trying to solve was that:

```
C:/boost/boost/assert/source_location.hpp(95,9): error C2039: '_snprintf': is not a member of 'std' (compiling source file <ANONYMIZED_PATH>/src/vision_opencv/cv_bridge/src/module_opencv3.cpp) [<ANONYMIZED_PATH>/build/cv_bridge/src/cv_bridge_boost.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\string(24): message: see declaration of 'std' (compiling source file <ANONYMIZED_PATH>/src/vision_opencv/cv_bridge/src/module_opencv3.cpp) [<ANONYMIZED_PATH>/build/cv_bridge/src/cv_bridge_boost.vcxproj]
C:/boost/boost/assert/source_location.hpp(95,9): error C2039: '_snprintf': is not a member of 'std' (compiling source file <ANONYMIZED_PATH>/src/vision_opencv/cv_bridge/src/module.cpp) [<ANONYMIZED_PATH>/build/cv_bridge/src/cv_bridge_boost.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\string(24): message: see declaration of 'std' (compiling source file <ANONYMIZED_PATH>/src/vision_opencv/cv_bridge/src/module.cpp) [<ANONYMIZED_PATH>/build/cv_bridge/src/cv_bridge_boost.vcxproj]
C:/boost/boost/assert/source_location.hpp(102,13): error C2039: '_snprintf': is not a member of 'std' (compiling source file <ANONYMIZED_PATH>/src/vision_opencv/cv_bridge/src/module_opencv3.cpp) [<ANONYMIZED_PATH>/build/cv_bridge/src/cv_bridge_boost.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\string(24): message: see declaration of 'std' (compiling source file <ANONYMIZED_PATH>/src/vision_opencv/cv_bridge/src/module_opencv3.cpp) [<ANONYMIZED_PATH>/build/cv_bridge/src/cv_bridge_boost.vcxproj]
C:/boost/boost/assert/source_location.hpp(102,13): error C2039: '_snprintf': is not a member of 'std' (compiling source file <ANONYMIZED_PATH>/src/vision_opencv/cv_bridge/src/module.cpp) [<ANONYMIZED_PATH>/build/cv_bridge/src/cv_bridge_boost.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\string(24): message: see declaration of 'std' (compiling source file <ANONYMIZED_PATH>/src/vision_opencv/cv_bridge/src/module.cpp) [<ANONYMIZED_PATH>/build/cv_bridge/src/cv_bridge_boost.vcxproj]
```

I can only solve this using -DHAVE_SNPRINTF

Reply:
Ok, i see where you may have gotten the idea from now (https://github.com/boostorg/system/issues/32#issuecomment-462912013).

Instead of defining the flag globally, could you use ``use target_compile_definitions`` to apply the flag just to the vision_opencv_boost target, and place this close to where the target is getting generated?

eg.

```
target_compile_definitions(${PROJECT_NAME}_boost PRIVATE HAVE_SNPRINTF)
```

somewhere around line 39 in **src/CmakeLists.txt``. (not this cmakelists.txt)